packages/flutter/test/foundation/diagnostics_json_test.dart

```diff
@@ -288,8 +291,9 @@ class TestDiagnosticsSerializationDelegate implements DiagnosticsSerializationDe
 
   @override
   DiagnosticsSerializationDelegate delegateForNode(DiagnosticsNode node) {
-    if (nodeDelegator != null) {
-      return nodeDelegator(node, this);
+    final NodeDelegator? localNodeDelegator = nodeDelegator;
```

Comment:
This function body could be replaced by:
```dart
    return nodeDelegator?.call(node, this) ?? (subtreeDepth > 0 ? copyWith(subtreeDepth: subtreeDepth - 1) : this);
```

Reply:
I think the current one is more readable (but I removed the local var and make it a bang).